runbotics/runbotics-orchestrator-ui/src/src-app/components/guards/AuthGuard.tsx

```diff
@@ -23,7 +26,8 @@ export const withAuthGuard = (Component: FC, featureKeys?: FeatureKey[], options
     const isAuthenticated = isInitialised && isBrowser && isAuthed;
 
     if (!isAuthenticated) {
-        router.replace('/');
+        if (appInURLRegex.test(router.asPath)) StorageManagerService.insertDestinationPage(router.asPath);
+        router.replace('/login');
```

Comment:
I would display a message to the user e.g. "Session expired, log in". Use enqueueSnackbar (see how it is use in other places in the app)